main.rb

```diff
@@ -1,17 +1,140 @@
 module Enumerable
-    def my_each
-        for i in 0...self.size
-            yield(self[i])
-        end
-    self
+  # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+
+  def my_each
+    i = 0
+    if block_given?
+      while i < length
+        yield(self[i])
+        i += 1
+      end
+      return self
+    end
+    to_enum
+  end
+
+  def my_each_with_index
+    i = 0
+    if block_given?
+      while i <= length - 1
+        yield(self[i], i)
+        i += 1
+      end
+      return self
     end
+    to_enum
+  end
+
+  def my_select
+    arr = []
+    return to_enum unless block_given?
+
+    my_each { |x| arr << x if yield(x) == true }
+    arr
+  end
 
-    def my_each_with_index
-        for i in 0...self.size
-            yield(self[i], i)
+  def my_all?(arg = nil, &prc)
+    if block_given?
+      my_each { |x| return false if prc.call(x) == false }
+    elsif arg.is_a?(Regexp)
+      my_each { |x| return false if arg.match?(x.to_s) == false }
+    elsif arg.is_a?(Class)
+      my_each { |x| return false if x.is_a?(arg) == false }
+    elsif arg.nil? == false
+      my_each { |x| return false if x != arg }
+    else
+      my_each { |x| return false if x.nil? == true }
+    end
+    true
+  end
+
+  def my_any?(&prc)
+    if block_given?
+      my_each { |x| return true if prc.call(x) }
+    else
+      my_each { |x| return true if x.nil? == false }
+    end
+    false
+  end
+
+  def my_none?(arg = nil, &prc)
+    if block_given?
+      my_each { |x| return false if prc.call(x) == true }
+    elsif arg.is_a?(Regexp)
+      my_each { |x| return false if arg.match?(x.to_s) == true }
+    elsif arg.is_a?(Class)
+      my_each { |x| return false if x.is_a?(arg) == true }
+    elsif arg.nil? == false
+      my_each { |x| return false if x == arg }
+    else
+      my_each { |x| return false if x.nil? == false }
```

Comment:
when no block or argument is given returns true only if none of the collection members is true. Here you are also comparing it with nil. What if we have a false? I think
```ruby
my_each { |x| return false if x.nil? == false }
```
should be 
```ruby
my_each { |x| return false if x }
```